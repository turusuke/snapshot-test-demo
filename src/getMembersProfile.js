export default function getMembersProfile(targetName) {
  return [
    {
      name: 'taro',
      age: 41
    },
    {
      name: 'hanako',
      age: 32
    },
    {
      name: 'jiro',
      age: 21
    }
  ].filter(({name}) => targetName ? targetName === name : true)
}
