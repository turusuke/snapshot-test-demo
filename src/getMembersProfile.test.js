import getMembersProfile from "./getMembersProfile";

describe("getMembersProfile", () => {
  it("全員のプロフィールを取得することができる", () => {
    expect(getMembersProfile()).toMatchSnapshot();
  });

  it("指定した人のプロフィールだけ取得することができる", () => {
    expect(getMembersProfile("hanako")).toMatchSnapshot();
  });
});

describe("getMembersProfile[unused snapshot]", () => {
  it("全員のプロフィールを取得することができる", () => {
    const expectedProfile = [
      {
        name: "taro",
        age: 41
      },
      {
        name: "hanako",
        age: 32
      },
      {
        name: "jiro",
        age: 21
      }
    ];
    expect(getMembersProfile()).toStrictEqual(expectedProfile);
  });

  it("指定した人のプロフィールだけ取得することができる", () => {
    const expectedProfile = [
      {
        name: "hanako",
        age: 32
      }
    ];
    expect(getMembersProfile("hanako")).toStrictEqual(expectedProfile);
  });
});

describe("getMembersProfile[used inlineSnapshot]", () => {
  it("全員のプロフィールを取得することができる", () => {
    expect(getMembersProfile()).toMatchInlineSnapshot(`
                  Array [
                    Object {
                      "age": 41,
                      "name": "taro",
                    },
                    Object {
                      "age": 32,
                      "name": "hanako",
                    },
                    Object {
                      "age": 21,
                      "name": "jiro",
                    },
                  ]
            `);
  });

  it("指定した人のプロフィールだけ取得することができる", () => {
    expect(getMembersProfile("hanako")).toMatchInlineSnapshot(`
      Array [
        Object {
          "age": 32,
          "name": "hanako",
        },
      ]
    `);
  });
});
