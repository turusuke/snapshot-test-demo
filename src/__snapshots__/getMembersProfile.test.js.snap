// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMembersProfile 全員のプロフィールを取得することができる 1`] = `
Array [
  Object {
    "age": 41,
    "name": "taro",
  },
  Object {
    "age": 32,
    "name": "hanako",
  },
  Object {
    "age": 21,
    "name": "jiro",
  },
]
`;

exports[`getMembersProfile 指定した人のプロフィールだけ取得することができる 1`] = `
Array [
  Object {
    "age": 32,
    "name": "hanako",
  },
]
`;
